src/main/java/com/iota/iri/model/Hash.java

```diff
@@ -147,7 +187,8 @@ public Indexable decremented() {
 
     @Override
     public int compareTo(Indexable indexable) {
-        Hash hash = new Hash(indexable.bytes());
+        Objects.requireNonNull(indexable);
```

Comment:
Just one last thing (didn't notice it before)...
Unlike the other places where you rightfully added an early null fail check, here it seems extraneous.
The next line will throw the exact same exception and it will be very clear that `indexable` will be the one to blame.

Java is a rather verbose language, but I don't think there is a need to over do it.